Mage/src/main/java/mage/abilities/condition/common/SourceHasRemainedOnBattlefieldCondition.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Copyright 2010 BetaSteward_at_googlemail.com. All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without modification, are
+ *  permitted provided that the following conditions are met:
+ *
+ *     1. Redistributions of source code must retain the above copyright notice, this list of
+ *        conditions and the following disclaimer.
+ *
+ *     2. Redistributions in binary form must reproduce the above copyright notice, this list
+ *        of conditions and the following disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY BetaSteward_at_googlemail.com ``AS IS'' AND ANY EXPRESS OR IMPLIED
+ *  WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ *  FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL BetaSteward_at_googlemail.com OR
+ *  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ *  SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ *  ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ *  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+ *  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *  The views and conclusions contained in the software and documentation are those of the
+ *  authors and should not be interpreted as representing official policies, either expressed
+ *  or implied, of BetaSteward_at_googlemail.com.
+ */
+package mage.abilities.condition.common;
+
+import java.util.UUID;
+import mage.abilities.Ability;
+import mage.abilities.condition.Condition;
+import mage.game.Game;
+
+
+/**
+ * As long as the sourceId permanent has
+ * remained on the battlefield, the condition is true.
+ *
+ * @author spjspj
+ */
+public class SourceHasRemainedOnBattlefieldCondition implements Condition {
+
+    UUID permanentId = null;
+    int timesChangedZones = -1;
+
+    public SourceHasRemainedOnBattlefieldCondition getInstance() {
+        return new SourceHasRemainedOnBattlefieldCondition();
+    }
+
+    public void setPermanentId(UUID permanentId) {
+        this.permanentId = permanentId;
+        timesChangedZones = -1;
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        if (permanentId != null && (game.getPermanent(permanentId) != null)) {
+            if (timesChangedZones == -1) {
+                timesChangedZones = game.getPermanent(permanentId).getZoneChangeCounter(game);
+            }
```

Comment:
So more general you can use:
  game.getState().getZoneChangeCounter(permanentId)